tools-local/tasks/ChangeEntryPointLibraryName.Netstandard2_0.cs

```diff
@@ -0,0 +1,245 @@
+
+        public string NewName { get; set; }
+
+        private string _version;
+
+        public override bool Execute()
+        {
+            bool retVal = false;
+
+            byte[] utf8Json = File.ReadAllBytes(DepsFile);
+
+            // TODO: Replace the use of the Utf8JsonReader with the JsonDocument once it is available.
+
+            var reader = new Utf8JsonReader(utf8Json, isFinalBlock: true, state: default);
+
+            using (var filestream = new FileStream(DepsFile, FileMode.Create, FileAccess.Write))
+            {
+                using (var bufferWriter = new StreamBufferWriter(filestream))
+                {
+                    var state = new JsonWriterState(options: new JsonWriterOptions { Indented = true });
+                    var jsonWriter = new Utf8JsonWriter(bufferWriter, state);
+
+                    retVal = reader.Read();
+
+                    if (retVal)
+                    {
+                        Debug.Assert(reader.TokenType == JsonTokenType.StartObject);
+                        jsonWriter.WriteStartObject();
+                        ReadWriteObject(ref reader, ref jsonWriter);
+                        jsonWriter.Flush();
+                    }
+                }
+            }
+
+            return retVal;
+        }
+
+        internal static void Skip(ref Utf8JsonReader jsonReader)
+        {
+            int depth = jsonReader.CurrentDepth;
+            if (jsonReader.TokenType == JsonTokenType.PropertyName)
+            {
+                jsonReader.Read();
+            }
+
+            if (jsonReader.TokenType == JsonTokenType.StartObject || jsonReader.TokenType == JsonTokenType.StartArray)
+            {
+                while (jsonReader.Read() && depth < jsonReader.CurrentDepth)
+                {
+                }
+            }
+        }
+
+        internal static Exception CreateUnexpectedException(ref Utf8JsonReader reader, string expected)
+        {
+            return new FormatException($"Unexpected character encountered, excepted '{expected}' " +
+                                       $"at line {reader.CurrentState._lineNumber} position {reader.CurrentState._bytePositionInLine}");
+        }
+
+        private void ReadWriteObject(ref Utf8JsonReader reader, ref Utf8JsonWriter writer)
+        {
+            while (reader.Read())
+            {
+                switch (reader.TokenType)
+                {
+                    case JsonTokenType.PropertyName:
+                        ReadWritePropertyName(reader.GetStringValue(), ref reader, ref writer);
+                        break;
+                    case JsonTokenType.EndObject:
+                        writer.WriteEndObject();
+                        break;
+                }
+            }
+        }
+
+        private void ReadWritePropertyName(string propertyName, ref Utf8JsonReader reader, ref Utf8JsonWriter writer)
+        {
+            if (propertyName == "targets")
```

Comment:
We need to use the `JsonDocument` here. The order in which the "targets" property and the "libraries" property show up could affect the end results here if we use the Reader and Writer. This relies on "targets" coming up before "libraries".

Reply:
https://dotnet.myget.org/feed/dotnet-core/package/nuget/Microsoft.Bcl.Json.Sources says JsonDocument is available now (4.6.0-preview.19073.1 is bigger than the previous version)